src/herbivoresAndCarnivores.js

```
@@ -2,14 +2,44 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+  isAlive() {
+    this.health = 0;
+    Animal.alive = Animal.alive.filter((a) => a.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
```

Behaviour to maintain: `hide()` currently sets `this.hidden = true`, which satisfies the checklist requirement that `hide` should set `hidden` to true (not toggle). If you change Animal.alive to store plain objects, remember to also set the corresponding plain-object entry's `hidden` property so the static array reflects the hidden state (e.g., `this.record.hidden = true`). The task examples expect `hidden: false` by default for newly created Herbivores and `hidden: true` after `hide()` is called .